pkg/zeek/addr.go

```diff
@@ -24,68 +25,85 @@ func (t *TypeOfAddr) String() string {
 	return "addr"
 }
 
-func (t *TypeOfAddr) Parse(value []byte) (net.IP, error) {
+func EncodeAddr(a net.IP) zval.Encoding {
+	ip := a.To4()
+	if ip == nil {
+		ip = net.IP(a)
+	}
+	return []byte(ip)
```

Comment:
For every `Encode*` I would also like to have an `AppendAddr(a net.IP, in zval.Encoding) zval.Encoding`. I also wouldn't be opposed to changing the `Encode*` args to `func EncodeAddr(zval.Encoding, net.IP) zval.Encoding` since you can just easily do `EncodeAddr(nil, ip)`.

Reply:
I would like to do the big merge then refine the API on a subsequent PR.  You have a good point and we might want both AppendThing and EncodeThing.  I also am harboring thoughts about making the core API around TypedEncoding and maybe even getting rid of the zeek.Value interface.  There are lots of optimizations we could do with TypedEncodings when compiling expressions/reducers into closures.

Reply:
Sounds good!